torch/csrc/distributed/c10d/TCPStore.cpp

```diff
@@ -236,6 +237,11 @@ TCPStore::TCPStore(std::string host, const TCPStoreOptions& opts)
     : Store{opts.timeout},
       addr_{std::move(host)},
       numWorkers_{opts.numWorkers} {
+  if (opts.useLibUV && !::c10d::detail::is_libuv_tcpstore_backend_available()) {
+    throw std::invalid_argument(
+        "use_libuv was requested but PyTorch was build without libuv support");
+  }
```

Comment:
Should probably do `TORCH_CHECK(::c10d::detail::is_libuv_tcpstore_backend_available(), "use_libuv was requested but PyTorch was build without libuv support");` instead

TORCH_CHECK goes through the pytorch exception flow and also logs the C++ stack trace

Reply:
fixed.